analytic_tag_dimension/models/analytic.py

```diff
@@ -0,0 +1,108 @@
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticDimension(models.Model):
+    _name = "account.analytic.dimension"
+    _description = "Account Analytic Dimension"
+
+    name = fields.Char(required=True)
+    code = fields.Char(required=True)
+    analytic_tag_ids = fields.One2many(
+        comodel_name="account.analytic.tag",
+        inverse_name="analytic_dimension_id",
+        string="Analytic Tags",
+    )
+
+    @api.model
+    def create(self, values):
+        if " " in values.get("code"):
+            raise ValidationError(_("Code can't contain spaces!"))
+        model_names = (
+            "account.move.line",
+            "account.analytic.line",
+            "account.invoice.report",
+        )
+        _models = self.env["ir.model"].search(
+            [("model", "in", model_names)], order="id"
+        )
+        _models.write(
+            {
+                "field_id": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "x_dimension_{}".format(values.get("code")),
+                            "field_description": values.get("name"),
+                            "ttype": "many2one",
+                            "relation": "account.analytic.tag",
+                        },
+                    )
+                ]
+            }
+        )
+        return super().create(values)
+
+
+class AccountAnalyticTag(models.Model):
+    _inherit = "account.analytic.tag"
+
+    analytic_dimension_id = fields.Many2one(
+        comodel_name="account.analytic.dimension", string="Dimension"
+    )
+
+    def get_dimension_values(self):
+        values = {}
+        for tag in self.filtered("analytic_dimension_id"):
+            code = tag.analytic_dimension_id.code
+            values.update({"x_dimension_%s" % code: tag.id})
+        return values
+
+    def _check_analytic_dimension(self):
+        tags_with_dimension = self.filtered("analytic_dimension_id")
+        dimensions = tags_with_dimension.mapped("analytic_dimension_id")
+        if len(tags_with_dimension) != len(dimensions):
+            raise ValidationError(_("You can not set two tags from same dimension."))
+
+
+class AnalyticDimensionLine(models.AbstractModel):
+    _name = "analytic.dimension.line"
+    _description = "Analytic Dimension Line"
+    _analytic_tag_field_name = "analytic_tag_ids"
+
+    def _handle_analytic_dimension(self):
+        for adl in self:
+            tag_ids = adl[self._analytic_tag_field_name]
+            tag_ids._check_analytic_dimension()
+            dimension_values = tag_ids.get_dimension_values()
+            super(AnalyticDimensionLine, adl).write(dimension_values)
+
+    def create(self, values):
+        result = super(AnalyticDimensionLine, self).create(values)
+        for value in values:
+            if value.get(result._analytic_tag_field_name):
+                result._handle_analytic_dimension()
```

Comment:
```suggestion
        if values.get(result._analytic_tag_field_name):
           result._handle_analytic_dimension()
```
you cannot iterate here, and actually don't need to

Reply:
@i-vyshnevska values are list in dict can't used get

Reply:
maybe you about create_multi decorator but in creating we have dict, not a list, current PR is broken 
https://github.com/OCA/account-analytic/blob/ee2001213f40bbdd6e9943a2d29045c6cf6b8e71/analytic_tag_dimension/models/analytic.py#L85

Reply:
OH. Sorry, It's my mistake that delete `@api.model` in function create. Fixed this file and this PR. Thank you for your advise :)

Reply:
@i-vyshnevska `@api.model` in v12 and v13 is difference, we can used `@api.model_create_multi` instead. However it's return dict in list, So I used `filter` function to check condition instead of for loops. What do you think?

Reply:
even if @api.model was changed we shouldn't fill any difference on this level so it should work with it as well
suppose you can use  model_create_multi 
will see on tests

Reply:
model_create_multi will return values is dict in list. So, at first I used for loop